x/profiles/keeper/keeper_chain_link.go

```diff
@@ -0,0 +1,121 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/desmos-labs/desmos/x/profiles/types"
+)
+
+// StoreChainLink stores the given chain link inside the current context.
+// It assumes that the given chain link has already been validated.
+func (k Keeper) StoreChainLink(ctx sdk.Context, user string, link types.ChainLink) error {
+
+	srcAddrData, err := types.UnpackAddressData(k.cdc, link.Address)
+	if err != nil {
+		return err
+	}
+
+	target := srcAddrData.GetAddress()
+	if _, found := k.GetAccountByChainLink(ctx, link.ChainConfig.Name, target); found {
+		return fmt.Errorf("chain link already exists")
+	}
+
+	// check target address has a profile or not
+	if link.ChainConfig.Name == "desmos" {
```

Comment:
Once the above mentioned constant is defined, we should use it here as well instead of the string value `"desmos"`

Reply:
I've thought more about this @dadamu and I think we should change this approach. Instead, what we can do without having to hard-code the Desmos chain name, is check the value of the address. A Desmos address will always be Bech32 encoded and we will always be able to read it as an `AccAddress`. For this reason, we could define the following method inside the `types` folder: 

```golang
func IsDesmosAddress(data AddressData) bool {
  // 1. Try reading the data as Bech32 address
  bech32data, ok := data.(*Bech32Address)
  if !ok {
    return false
  }
  
  // 2. Try decoding the data as an AccAddress
  _, err := sdk.AccAddressFromBech32(data.Value) 
  return err != nil
}
```

Note: the code needs to be changed to do the cast properly, but it should give you an idea on how to check this

Reply:
Thinking more about this approach, we might have some problems with Base58-encoded addresses. I think we might want to have only Bech32-encoded ones to avoid two profiles being linked together 🤔 Otherwise we won't be able to make sure the encoded address is not associated with any profile, and users might be able to trick the system by setting a different chain name or using a different encoding

Reply:
Another idea: we simply ditch the check on address and profiles all together and allow any link being created. We should have no problems in doing so since the links are stored inside each profile as a list of addresses. 

Consider we have profile `A` and profile `B`. Then if we link `A -> B` we will see inside `A` the address of `B`, but not the other way around. If we want to see the address of `A` inside `B` then we will have to create another link `B -> A`. 

This way we only keep a one-level link between different accounts and profiles, the same we do with relationships inside the `x/posts` module. What do you guys think @bragaz @dadamu?

Reply:
@RiccardoM True, the Base58-encoded might have some problems cause there is no method to check the validity of address.

Imagine that the user want to create a link with wrong type encoded address `base58address` from `cosmos`. Then, The create link process will success. However, other users will know it is invalid address cause `cosmos` address should be `bech32` encoded and have `cosmos` prefix. As the result, the user only wastes gas fee to create an useless link.

Reply:
@dadamu I really think we should avoid any kind of check and let users create the links with any valid Bech32 or Base58-encoded values. If they want to waste some gas and fees creating wrong links it will be their fault. 

We should also delete the check and avoid two profiles being linked together In my opinion. It won't cause any problem and it will simplify a lot the logic of the code.

Reply:
@RiccardoM I agree with it. The multiple profile does not cause any problem. With it, Should `QueryProfileByChainLink` be replaced with `QueryProfilesByChainLink` to show multiple profiles the address is associated?

Reply:
@dadamu No, we should always keep one external link associated to a single profile, so the query can stay the same. Using that query, we only return the 1-level links we have, not multiple levels. 

Consider the following situation: 

![Untitled Diagram(5)](https://user-images.githubusercontent.com/9119016/120754939-9e89a680-c50d-11eb-87bb-9cfc3c14deef.png)

If we query using the `cosmos` address, we only return the `Desmos 1` profile, not the `Desmos 2`. We only care about the first level links

Reply:
@RiccardoM I got it. Just allow `desmos` address which has profile already to link to another profile. And querying profile by chain link returns linked profile instead of itself. And keep every address has only one link.